app/test/unit/git/format-patch-test.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  setupTwoCommitRepo,
+  setupFixtureRepository,
+} from '../../helpers/repositories'
+import { Repository } from '../../../src/models/repository'
+import { formatPatch } from '../../../src/lib/git'
+import {
+  cloneLocalRepository,
+  makeCommit,
+} from '../../helpers/repository-scaffolding'
+import { GitProcess } from 'dugite'
+
+describe('formatPatch', () => {
+  describe('in a repo with commits', () => {
+    let repository: Repository
+    beforeEach(async () => {
+      repository = await setupTwoCommitRepo()
+      await makeCommit(repository, {
+        entries: [{ path: 'another-one', contents: 'dusty' }],
+      })
+    })
+    it('returns a string for a single commit range', async () => {
+      const patch = await formatPatch(repository, 'HEAD~', 'HEAD')
+      expect(patch).toBeString()
+      expect(patch).not.toBeEmpty()
+    })
+    it('returns a string for a multi commit range', async () => {
+      const patch = await formatPatch(repository, 'HEAD~~', 'HEAD')
+      expect(patch).toBeString()
+      expect(patch).not.toBeEmpty()
+    })
+    it('returns empty string for no range', async () => {
+      const patch = await formatPatch(repository, 'HEAD', 'HEAD')
+      expect(patch).toBeString()
+      expect(patch).toBeEmpty()
+    })
+    describe('applied in a related repo', () => {
+      let clonedRepository: Repository
+      beforeEach(async () => {
+        clonedRepository = await cloneLocalRepository(repository)
+        await makeCommit(clonedRepository, {
+          entries: [{ path: 'okay-file', contents: 'okay' }],
+        })
+      })
+      it('is valid', async () => {
```

Comment:
:art: to make this test a bit more descriptive of what Git is doing with the patch

```suggestion
      it('will be applied cleanly', async () => {
```